src/late/webui/preload.js

```diff
@@ -31,6 +35,37 @@ window.ipfsDesktop = {
 
   toggleSetting: (setting) => {
     ipcRenderer.send('config.toggle', setting)
+  },
+
+  selectDirectory: () => {
+    return new Promise(resolve => {
+      remote.dialog.showOpenDialog(remote.getCurrentWindow(), {
+        title: 'Select a directory',
+        properties: [
+          'openDirectory',
+          'createDirectory'
+        ]
+      }, async (res) => {
+        if (!res || res.length === 0) {
+          return resolve()
```

Comment:
Would it be useful to make that a reject, so calling code can tell when you've cancelled? Testing out this PR the current behaviour if you hit cancel from the file dialog it leaves the drop down menu open, but if you cancel from the directory dialog it closes the drop down. I think it should close the drop down in both cases.

Reply:
The issue's on Web UI code. I'll apply a fix on Web UI soon. We shouldn't reject I think because on the browser side we just receive a `null` variable, not a rejection.

Reply:
See https://github.com/ipfs-shipyard/ipfs-webui/pull/1051